backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java

```diff
@@ -600,9 +604,16 @@ private ExperimentRun populateFieldsBasedOnPrivileges(ExperimentRun experimentRu
 
   private ExperimentRun checkDatasetVersionBasedOnPrivileges(ExperimentRun experimentRun) {
     ExperimentRun.Builder experimentRunBuilder = experimentRun.toBuilder();
+    List<Artifact> accessibleDatasetVersions =
+        getPrivilegedDatasets(experimentRun.getDatasetsList());
+    experimentRunBuilder.clearDatasets().addAllDatasets(accessibleDatasetVersions);
+    return experimentRunBuilder.build();
+  }
+
+  private List<Artifact> getPrivilegedDatasets(List<Artifact> newArtifacts) {
     List<Artifact> accessibleDatasetVersions = new ArrayList<>();
     List<String> accessibleDatasetVersionIds = new ArrayList<>();
-    for (Artifact dataset : experimentRun.getDatasetsList()) {
+    for (Artifact dataset : newArtifacts) {
```

Comment:
make the variable name consistent.
IF all ids are datasetVersionId then let us call them datasetVersion instead of artifacts and dataset